hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/InternalExecutionService.java

```diff
@@ -18,14 +18,27 @@
 
 import com.hazelcast.spi.ExecutionService;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
 /**
  * The internal {@link ExecutionService}. Currently this method doesn't contains any additional methods, but
  * in the future they can be added here.
  */
 public interface InternalExecutionService extends ExecutionService {
 
+    ExecutorService getDurable(String name, boolean scheduled);
 
     void executeDurable(String name, Runnable command);
 
+    ScheduledFuture<?> scheduleDurable(String name, Runnable command, long delay, TimeUnit unit);
+
+    <V> ScheduledFuture<V> scheduleDurable(String name, Callable<V> command, long delay, TimeUnit unit);
+
+    ScheduledFuture<?> scheduleDurableWithRepetition(String name, Runnable command,
```

Comment:
Regarding the reasons behind the method selection, is that the implementation was already supported so it was much more trivial to rely on that. I am afraid, that changing either at this stage is a much bigger change.  One trivial fix which I don't know if it will have any side effects, would be to change the order of the decorators before scheduling the task, ie. 
`Runnable decoratedTask = new SkipOnConcurrentExecutionDecorator(new DelegatingTaskDecorator(command, executor));` thus avoiding further queuing of the task if it is still running.

@jerrinot any opinions ?